C++/Number of distinct characters in string.cpp

```diff
@@ -0,0 +1,16 @@
+#include <iostream>
+#include <map>
+using namespace std;
+int main() {
+    string s;
+    int count = 0;
+    map<char,int> mark;
+    cin >> s;
+    for(int i = 0; i < s.size(); i++){
+        if(!mark[s[i]]){
+            mark[s[i]] = 1;
+            count++;
+        }
+    }
+    cout << "There are " << count << " distinct characters." << endl;
```

Comment:
Code logic incorrect 
- Unnecessary space used
- use of map is incorrect
- code doesn't work for a string with duplicate characters(fails the only requirement)